hydroshare_jupyter_sync/lib/filesystem/fs_resource_map.py

```diff
@@ -0,0 +1,199 @@
+from abc import ABC, abstractmethod
+from collections import UserDict
+from typing import Dict, List, Union
+from hsclient import Resource
+import hashlib
+from pathlib import Path
+
+from .types import MD5Hash
+
+
+def compute_file_md5_hexdigest(file: Union[str, Path]) -> str:
+    """Compute a file's md5 hexdigest. Read file as chunks to conserve memory usage."""
+    with open(file, "rb") as f:
+        hash = hashlib.md5()
+        chunk = f.read(8192)
+        while chunk:
+            hash.update(chunk)
+            chunk = f.read(8192)
+
+        return hash.hexdigest()
```

Comment:
I've never written an md5 checksum function but this seems about right.  HydroShare uses the value irods computes.  If you haven't already, please ensure Hydroshare gets the same checksum as what is computed here.

Reply:
There is not a unit test that verifies this, but I have checked that this implementation matches HydroShare's results.